### DISCUSSION TITLE
Package variant related query

### DISCUSSION THREAD

**Discussion by SamarSidharth** [kptdev/kpt#4110] [General]
I have created a packagevariant and I want to test the values I can update in it after successfull creation
I update the configMap value `put-value` field used to pass parameter to a kpt function under pipeline.mutator and got the below errors

```shell
  pipeline:
    mutators:
    - configMap:
        by-path: spec.kubeConfig.secretRef.name
        put-value: cluster2-kubeconfig
      image: gcr.io/kpt-fn/search-replace:v0.2.0
      selectors:
      - kind: HelmRelease

```

I get the below error in controller

```shell
I0103 13:07:41.612703       1 packagevariant_controller.go:819] PackageVariant "nrf-free5gc-nif", PackageRevision "downstream-nif-6daa2a1d959b8eda03c4923f85b9465daa18b3a7", resources changed: "Kptfile" different
I0103 13:07:41.612731       1 packagevariant_controller.go:413] package variant "nrf-free5gc-nif" needs to mutate to package revision "downstream-nif-6daa2a1d959b8eda03c4923f85b9465daa18b3a7", creating new draft
I0103 13:07:41.612743       1 packagevariant_controller.go:637] package variant "nrf-free5gc-nif" is creating package revision ""
E0103 13:07:41.672844       1 packagevariant_controller.go:417] package variant "nrf-free5gc-nif" failed to copy "downstream-nif-6daa2a1d959b8eda03c4923f85b9465daa18b3a7": Internal error occurred: error applying patch: conflict: fragment line does not match src line
```

**Comment by johnbelamaric** [comment]
That's a definite bug. Are you able to share the upstream package? That error is for the low-level git merge code.

**Reply by SamarSidharth** [reply]
I created an upstream kpt package and then created a package variant to create a downstream package from it which was successfully created with all pipeline and injectors in package variant.

The scenario I wanted test was to manually edit few values in pipeline of existing package variant to see if I can generate a new downstream package in draft.


Not sure if I answered your query about sharing the upstream package

**Reply by johnbelamaric** [reply]
Well, I was asking for the actual package to see if we can reproduce this. Also what version of Porch? Does it happen every time or is it only occasional? I wonder if this is a concurrency issue - a partial edit getting saved. Or maybe something specific with the search-replace where it is thwacking something it shouldn't? This should in theory "never happen"...

**Reply by SamarSidharth** [reply]
I can share the packages.. also I will try to test more to see the frequency of this issue. 
I got the same error message when I tried to update PV by making changes to the injector config map

Porch version: v0.0.31

**Reply by SamarSidharth** [reply]
PFA the upstream package
[nrf_2.zip](https://github.com/kptdev/kpt/files/13827559/nrf_2.zip)

**Reply by SamarSidharth** [reply]
I tested it multiple and I am facing this issue everytime. I tried to update the injected resource this time
```shell
I0104 14:31:22.630383       1 packagevariant_controller.go:819] PackageVariant "nrf-free5gc-nif", PackageRevision "downstream-nif-6daa2a1d959b8eda03c4923f85b9465daa18b3a7", resources changed: "nrf-overrides-cm.yaml" different
I0104 14:31:22.630503       1 packagevariant_controller.go:413] package variant "nrf-free5gc-nif" needs to mutate to package revision "downstream-nif-6daa2a1d959b8eda03c4923f85b9465daa18b3a7", creating new draft
I0104 14:31:22.630588       1 packagevariant_controller.go:637] package variant "nrf-free5gc-nif" is creating package revision ""
E0104 14:31:22.711759       1 packagevariant_controller.go:417] package variant "nrf-free5gc-nif" failed to copy "downstream-nif-6daa2a1d959b8eda03c4923f85b9465daa18b3a7": Internal error occurred: error applying patch: conflict: fragment line does not match src line

```

**Comment by johnbelamaric** [comment]
Porch has moved to Nephio - I created a bug there: https://github.com/nephio-project/nephio/issues/469

**Reply by SamarSidharth** [reply]
Thanks @johnbelamaric
